ee/performance/stats.go

```diff
@@ -0,0 +1,114 @@
+package performance
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/shirou/gopsutil/v4/process"
+)
+
+/*
+MemInfo is a helper struct with memory stats derived from both:
+- process.MemoryInfoStat (e.g. RSS, VMS). these help to give a picture of the usage from the OS perspective
+- runtime.MemStats (e.g. heap stats). these help to give a picture of the go runtime usage
+Used together, we are able to estimate things like allocations outside of our golang runtime (see nonGoMemUsage).
+The runtime MemStats can be confusing - here is an attempt at outlining some of the fields we're interested in:
+
+	Sys is total bytes of memory obtained from the OS. The virtual address space reserved by the Go runtime for:
+	Sys                             = HeapSys + StackSys + MSpanSys + MCacheSys + BuckHashSys + GCSys + OtherSys
+		HeapSys                     = HeapInUse + HeapIdle. estimates the largest size the heap has had
+			HeapInUse               = HeapAlloc + currently unused memory that has been dedicated to particular size classes
+				HeapAlloc           = bytes allocated for heap objects currently inuse + unreachable and pending GC
+			HeapIdle                = HeapReleased + bytes that are unused but have not yet been returned to OS
+				HeapReleased        = bytes of physical memory returned to the OS
+				...
+		...
+
+see https://github.com/golang/go/issues/32284#issuecomment-496967090 and https://pkg.go.dev/runtime#MemStats for further reading
+*/
+type MemInfo struct {
+	RSS           uint64  `json:"rss_bytes"`        // bytes - memory allocated to the process held in RAM. includes stack and heap memory
+	VMS           uint64  `json:"vms_bytes"`        // bytes - all memory the process can access, including allocated+unused, and from shared libraries
+	HeapTotal     uint64  `json:"heap_total_bytes"` // bytes - see heapTotal for details
+	GoMemUsage    uint64  `json:"go_mem_bytes"`     // bytes - see goMemUsage for details
+	NonGoMemUsage uint64  `json:"non_go_mem_bytes"` // bytes - see nonGoMemUsage for details
+	MemPercent    float32 `json:"mem_percent"`      // percent of memory in use (RSS) vs available on machine
+}
+
+type PerformanceStats struct {
+	Pid        int      `json:"pid"`
+	Exe        string   `json:"exe"`
+	MemInfo    *MemInfo `json:"mem_info"`
+	CPUPercent float64  `json:"cpu_percent"`
+}
+
+func CurrentProcessStats(ctx context.Context) (*PerformanceStats, error) {
```

Comment:
One caveat here, is that this gathers stats for the current process. Which _omits_ any desktop processes, or any osquery processes.